Sources/SwiftParser/Declarations.swift

```diff
@@ -1601,9 +1587,11 @@ extension Parser {
 
     let indices = self.parseParameterClause(for: .indices)
 
+    let unexpectedBeforeArrow = RawUnexpectedNodesSyntax(parseAllEffectsSpecifiers(), arena: self.arena)
+
     let result: RawReturnClauseSyntax
-    if self.at(.arrow) {
-      result = self.parseFunctionReturnClause().returnClause
+    if let handle = self.canRecoverTo(.arrow) {
+      result = self.parseReturnClause(handle).returnClause
```

Comment:
You are dropping the effect specifiers here, right?

Reply:
Weird that I can't see this in the review. But no, they're added to the return clause itself. They're only returned to be able to make `async/throws` missing when necessary. This is one of the many reasons I'm not a huge fan of this patch (though this particular returning of unexpected nodes was already here).

It really seems like we should just recover + place tokens in unexpected + have ASTGen handle these diagnostics. Though one issue there is that they wouldn't be parsed as contextual keywords if we just did regular recovery vs looked for the effects specifically.

`@unknown` is sort of an example of this, though it currently allows any attribute as the `@unknown` attribute and parses the rest as unexpected (ie. just recovers to the case/default).

Reply:
Ah, never mind. I missed that `throws` also gets added to the unexpected tokens of the `returnClause`.

Reply:
Yeah, but it's super easy to miss. Hence my problem with doing it this way :)